pages/api/comments/[id].js

```diff
@@ -0,0 +1,59 @@
+import dbConnect from "@/db/connect";
+import Post from "@/db/models/Post";
+import User from "@/db/models/User";
+import Comment from "@/db/models/Comment";
+
+export default async function handler(request, response) {
+  await dbConnect();
+  const { id } = request.query;
+
+  // adding a new comment
+  if (request.method === "PUT") {
+    const comment = await Comment.create(request.body);
+
+    //add a referece to the post
+    await Post.findByIdAndUpdate(
+      { _id: id },
+      { $push: { comments: comment._id } }
+    );
+
+    //add a referece to the user
+    await User.findByIdAndUpdate(
+      { _id: comment.user },
+      { $push: { comments: comment._id } }
+    );
```

Comment:
You have multiple database write / update operations in a row here. It is dangerous because if there is an exception (or the server crashes) after a subset of the operations you have inconsistent data in your database. In other words: Either all of these operations should be executed or none.
You can achieve this behavior using mongoose transactions:
```
const conn = await dbConnect();

const session = await conn.startSession();
const someReturnValue = await session.withTransaction(async () => {
     // All your operations here including a return statement (optional)
});
session.endSession();
```

Reply:
thank you @ThomasF85, but what do I do with the `someReturnValue` variable? 
Do I really need it? Isn't it possible to write just the following?

```suggestion
    const conn = await dbConnect();

const session = await conn.startSession();
await session.withTransaction(async () => {
     // All your operations here including a return statement (optional)
});
session.endSession();
```

Reply:
That's perfectly fine. The return value is optional in case you need it (for example if you want to send a newly created or updated object to the client)